crates/oxc_ast/src/serialize.rs

```diff
@@ -728,6 +728,40 @@ impl ESTree for TSModuleDeclarationGlobal<'_, '_> {
     }
 }
 
+/// Serializer for `body` field of `TSEnumDeclaration`.
+/// This field is derived from `members` field.
+///
+/// `Span` indicates within braces `{ ... }`.
+/// ```ignore
+/// enum Foo { ... }
+/// |              | TSEnumDeclaration.span
+///          |     | TSEnumBody.span
+///        ^^ id_end + 1 for space
+/// ```
+#[ast_meta]
+#[estree(
+    ts_type = "TSEnumBody",
+    raw_deser = "
+        const tsEnumDeclId = DESER[BindingIdentifier](POS_OFFSET.id);
+        const tsEnumDeclMembers = DESER[Vec<TSEnumMember>](POS_OFFSET.members);
+        const bodyStart = tsEnumDeclId.end + 1;
```

Comment:
I think this will work, and is more efficient as it only deserializes `id` field once instead of twice:

```suggestion
        const tsEnumDeclMembers = DESER[Vec<TSEnumMember>](POS_OFFSET.members);
        const bodyStart = THIS.id.end + 1;
```

Reply:
Sorry the `raw_deser` syntax is so dreadful! I will try to improve it at some point...